### PR TITLE
👌 Expand support for Python-Markdown in the admon plugin

### LINKED ISSUE #93: Admonition title syntax is not consistent with other implementation
### Describe the bug

**context**

The [admonition syntax](https://mdit-py-plugins.readthedocs.io/en/latest/#containers) is based on [Python Markdown](https://python-markdown.github.io/extensions/admonition/):

~~~markdown
!!! danger "Don't try this at home"
    ...
~~~

**expectation**

I would expect the quotes to be absent from the output.

**bug**

With mdit-py-plugins, the quotes are present in the output:

image

Unexpected output generated by mdit-py-plugins (spurious quotes):

~~~html
<div class="admonition danger"> <p class="admonition-title">"Don't try this at home"</p> <p>...</p> </div>
~~~

Moreover, the [additional CSS classes](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) are not supported either.

Changing the behavior might cause compatiblity issues with [markdown-it-admon](https://github.com/commenthol/markdown-it-admon) however.

**problem**

This is a problem for people migrating from Python Markdown.

### Reproduce the bug

Create markdown file with:

~~~markdown
!!! danger "Don't try this at home"
    ...
~~~

Create script

~~~python3
from markdown_it import MarkdownIt
from mdit_py_plugins.admon import admon_plugin

md_opts = {
    "html": True,
}

markdown_it_render = MarkdownIt("commonmark", md_opts).use(admon_plugin)

source = open("/dev/stdin", "rt").read()
res = markdown_it_render.render(source)
print(res)
~~~

Run:

~~~sh
python3 test.py < test.md
~~~

Output:

~~~html
<div class="admonition danger">
<p class="admonition-title">&quot;Don't try this at home&quot;</p>
<p>...</p>
</div>
~~~

### List your environment

* Python 3.11.4
* Debian Linux
* markdown-it-py==3.0.0
* mdit-py-plugins==0.4.0

### PR DESCRIPTION
**Pull request by KyleKing**
Fixes #93 

- Strip redundant quotes to improve support for [Python Markdown](https://python-markdown.github.io/extensions/admonition/)
- Follow Python-Markdown's syntax to parse the title in double quotes and use all other tokens as classes
- ~~Added missing HTML roles as identified by [#93](https://github.com/executablebooks/mdit-py-plugins/issues/93)~~ (removed after clarification in [a comment below](https://github.com/executablebooks/mdit-py-plugins/pull/94#discussion_r1237821692))

### CODE REVIEW

File: mdit_py_plugins/admon/index.py [hunk 1]
```diff
@@ -130,7 +130,10 @@ def admonition(state: StateBlock, startLine: int, endLine: int, silent: bool) ->
     token = state.push("admonition_open", "div", 1)
     token.markup = markup
     token.block = True
-    token.attrs = {"class": " ".join(["admonition", tag, *_extra_classes(markup)])}
+    token.attrs = {
+        "class": " ".join(["admonition", tag, *_extra_classes(markup)]),
+        "role": tag,
```
Comment (randomstuff) [inline mdit_py_plugins/admon/index.py:135]: `role` is for ARIA role. We should not propagate typical values such as `warning` there.

Comment (randomstuff) [reply mdit_py_plugins/admon/index.py:135]: … though it would make sense to set the `note` role for `!!! note`.

Comment (KyleKing) [reply mdit_py_plugins/admon/index.py:135]: Was just about to ask. So should all be `"note"` then (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/note_role)? I couldn't find how `role` was added in https://github.com/Python-Markdown/markdown for the example you provided in #93

Comment (randomstuff) [reply mdit_py_plugins/admon/index.py]: Sorry the role was an error on my part. I have some filter which adds some things on top of the raw markdown output :)

Comment (KyleKing) [reply mdit_py_plugins/admon/index.py]: No problem, I reverted the commits, so this should be ready with the fix for the quotes

File: tests/fixtures/admon.md [hunk 2]
```diff
@@ -53,6 +53,32 @@ Shows no title
...
+Parse additional classes to support Python markdown (https://github.com/executablebooks/mdit-py-plugins/issues/93#issuecomment-1601822723)
+.
+!!! a b c d inline-classes   "Note: note about "foo"" 
+    ...
+
+.
+<div class="admonition a b c d inline-classes">
+<p class="admonition-title">Note: note about &quot;foo&quot;</p>
+<p>...</p>
+</div>
```
Comment (KyleKing) [inline tests/fixtures/admon.md:78]: I also added a unit test. Feel free to recommend any tests that capture edge cases which we didn't have before

File: mdit_py_plugins/admon/index.py [hunk 3]
```diff
@@ -14,20 +17,34 @@
...
 
-def _get_tag(params: str) -> tuple[str, str]:
+def _get_multiple_tags(params: str) -> Tuple[List[str], str]:
+    """Check for multiple tags when the title is double quoted."""
+    re_tags = re.compile(r'^\s*(?P<tokens>[^"]+)\s+"(?P<title>.*)"\S*$')
+    match = re_tags.match(params)
+    if match:
+        tags = match["tokens"].strip().split(" ")
+        return [tag.lower() for tag in tags], match["title"]
+    raise ValueError("No match found for parameters")
```
Comment (KyleKing) [inline mdit_py_plugins/admon/index.py:27]: @randomstuff I liked your implementation in #95! I made some minor changes to prevent breaking backward compatibility

**Review by KyleKing — COMMENTED** [review]
Merged changes from #95

File: .coveragerc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+[report]
+exclude_lines =
+    pragma: no cover
+    if TYPE_CHECKING:
```
Comment (KyleKing) [inline .coveragerc:4]: I noticed that `codecov` was counting `TYPE_CHECKING` blocks as uncovered, but they can be auto-ignored with this minimal configuration

Without this change, code coverage is still increased, so this change could be extracted for a separate PR

---

Reference:

- Advice from 2019 should still be relevant: https://github.com/nedbat/coveragepy/issues/831#issuecomment-517778185
- Pinned documentation from latest release: https://github.com/nedbat/coveragepy/blob/a5f37e0ba1d00329d4e8aa68e3d17478feca157c/doc/excluding.rst#L148
- Live documentation: https://coverage.readthedocs.io/en/7.2.7/config.html#config

**Comment by randomstuff** [comment]
Looks good to me. :+1:

**Comment by KyleKing** [comment]
@chrisjsewell, this is a backward compatible change that is ready for review when you have a chance

**Comment by KyleKing** [comment]
Just bumping this PR! It is still ready for review/merge if there are no concerns

**Comment by KyleKing** [comment]
A friendly bump! This PR is ready for review and merge.

@chrisjsewell, I would like to merge this PR when possible because I am working on adding support for a new subvariant of admonitions that would follow these changes (See https://github.com/KyleKing/mdformat-admon/issues/14 and https://github.com/KyleKing/mdformat-admon/issues/17 for [Content Tabs](https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#usage))

**Review by chrisjsewell — APPROVED** [review]
Looks all good cheers!
